test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -258,15 +339,50 @@ private void CopyLatestCliToTestDirectory(string destinationDir)
 
             var lastWriteTime = DateTime.Now.AddDays(-2);
 
-            //Copy files recursively to destination directories
-            foreach (var fileName in Directory.GetFiles(cliDir, "*", SearchOption.AllDirectories))
+            //Copy files recursively to destination directories, ignoring any SDK version not selected.
+            foreach (var fileName in Directory.GetFiles(cliDir, "*", SearchOption.AllDirectories).Where(predicate))
             {
                 var destFileName = destinationDir + fileName.Substring(cliDir.Length);
                 File.Copy(fileName, destFileName);
                 File.SetLastWriteTime(destFileName, lastWriteTime);
             }
         }
 
+        private string GetSdkToTest(string sdkDir)
+        {
+            // The TFM we're testing
+            var testTfm = AssemblyReader.GetTargetFramework(typeof(MsbuildIntegrationTestFixture).Assembly.Location);
+
+            var selectedVersion =
+                Directory.GetDirectories(sdkDir) // get all directories in sdk folder
+                .Where(path =>
+                { // SDK is for TFM to test
+                    if (string.Equals(Path.GetFileName(path), "NuGetFallbackFolder", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return false;
+                    }
+
+                    var dotnetPath = Path.Combine(path, "dotnet.dll");
+                    var sdkTfm = AssemblyReader.GetTargetFramework(dotnetPath);
+
+                    return testTfm == sdkTfm;
+                })
+                .Select(Path.GetFileName) // just the folder name (version string)
+                .OrderByDescending(path => NuGetVersion.Parse(Path.GetFileName(path))) // in case there are multiple matching SDKs, selected the highest version
+                .FirstOrDefault();
```

Comment:
The `OrderByDescending` could not sort the semver correctly if there are preview and non-preview version with the same prefix.
But considering it's impossible to have both GA version and preview version with the same prefix on CI, I think it's acceptable.

Reply:
Interestingly, I couldn't find any tests that assert that a NuGetVersion without a prerelease label is sorted higher than a NuGetVersion with the same major, minor and patch values, but with a prerelease label.

However, [here's the code](https://github.com/NuGet/NuGet.Client/blob/4e31692ae65d3abc862791c2f4bdfc472ea00e32/src/NuGet.Core/NuGet.Versioning/VersionComparer.cs#L178-L188). Also, if we didn't sort this correctly, then it would be a violation of the Semantic Versioning 2 spec, so we'd have a bug. And Package Manager UI wouldn't sort packages in the correct order. For example, [check the version history of one of our NuGet packages on nuget.org](https://www.nuget.org/packages/NuGet.Versioning/), and see how the 5.6.0 release package is higher than the 5.6.0-preview packages. I'm not sure if NuGetGallery is using NuGetVersion to sort their list, but you can always check the packages in PM UI.

Reply:
Thanks for clarification! 
Sorry, I tested with string yesterday, so the preview version is higher. Just now I tested with NuGet.Version, and it works correctly.